barretenberg/cpp/src/barretenberg/stdlib/recursion/goblin/full_goblin_recursion.test.cpp

```diff
@@ -0,0 +1,95 @@
+#include "barretenberg/proof_system/circuit_builder/ultra_circuit_builder.hpp"
+#include "barretenberg/stdlib/recursion/honk/verifier/ultra_recursive_verifier.hpp"
+#include "barretenberg/translator_vm/goblin_translator_composer.hpp"
+#include "barretenberg/ultra_honk/ultra_composer.hpp"
+
+#include <gtest/gtest.h>
+
+using namespace proof_system::honk;
+namespace goblin_recursion_tests {
+
+class GoblinRecursionTests : public ::testing::Test {
+  protected:
+    static void SetUpTestSuite()
+    {
+        barretenberg::srs::init_crs_factory("../srs_db/ignition");
+        barretenberg::srs::init_grumpkin_crs_factory("../srs_db/grumpkin");
+    }
+
+    using Curve = curve::BN254;
+    using FF = Curve::ScalarField;
+    using Fbase = Curve::BaseField;
+    using Point = Curve::AffineElement;
+    using CommitmentKey = pcs::CommitmentKey<Curve>;
+    using OpQueue = proof_system::ECCOpQueue;
+    using GoblinUltraBuilder = proof_system::GoblinUltraCircuitBuilder;
+    using ECCVMFlavor = flavor::ECCVM;
+    using ECCVMBuilder = proof_system::ECCVMCircuitBuilder<ECCVMFlavor>;
+    using ECCVMComposer = ECCVMComposer_<ECCVMFlavor>;
+    using TranslatorFlavor = flavor::GoblinTranslator;
+    using TranslatorBuilder = proof_system::GoblinTranslatorCircuitBuilder;
+    using TranslatorComposer = GoblinTranslatorComposer;
+    using TranslatorConsistencyData = barretenberg::TranslationEvaluations;
+    using Proof = proof_system::plonk::proof;
+    using NativeVerificationKey = flavor::GoblinUltra::VerificationKey;
+    using RecursiveFlavor = flavor::GoblinUltraRecursive_<GoblinUltraBuilder>;
+    using RecursiveVerifier = proof_system::plonk::stdlib::recursion::honk::UltraRecursiveVerifier_<RecursiveFlavor>;
+    using KernelInput = Goblin::AccumulationOutput;
+
+    /**
+     * @brief Construct a mock kernel circuit
+     * @details This circuit contains (1) some basic/arbitrary arithmetic gates, (2) a genuine recursive verification of
+     * the proof provided as input. It does not contain any other real kernel logic.
+     *
+     * @param builder
+     * @param kernel_input A proof to be recursively verified and the corresponding native verification key
+     */
+    static void construct_mock_kernel_circuit(GoblinUltraBuilder& builder, KernelInput& kernel_input)
+    {
+        // Generic operations e.g. state updates (just arith gates for now)
+        GoblinTestingUtils::construct_arithmetic_circuit(builder);
+
+        // Execute recursive aggregation of previous kernel proof
+        RecursiveVerifier verifier{ &builder, kernel_input.verification_key };
+        // TODO(https://github.com/AztecProtocol/barretenberg/issues/801): Aggregation
+        [[maybe_unused]] auto pairing_points = verifier.verify_proof(kernel_input.proof);
+    }
+};
+
+/**
+ * @brief A full Goblin test that mimicks the basic aztec client architecture
+ *
+ */
+TEST_F(GoblinRecursionTests, Pseudo)
+{
+    barretenberg::Goblin goblin;
```

Comment:
Appreciate the `X x;` and `X x {y};` constructor usage, I find it nice to only see auto when it's actually needed